File: src/main/java/be/breina/openrgb/model/ObjectFactory.java
package be.breina.openrgb.model;

import be.breina.openrgb.util.LittleEndianInputStream;
import java.io.IOException;
import java.io.InputStream;

public final class ObjectFactory {

    public static Device device(InputStream inputStream) {
        try (var input = new LittleEndianInputStream(inputStream)) {
            return new Device(input);

        } catch (IOException e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: src/main/kotlin/be/breina/openrgb/model/ObjectFactory.kt
package be.breina.openrgb.model

import be.breina.openrgb.util.LittleEndianInputStream
import java.io.IOException
import java.io.InputStream

object ObjectFactory {
    fun device(inputStream: InputStream): Device? {
        try {
            LittleEndianInputStream(inputStream).use { input -> return Device.from(input) }
        } catch (e: IOException) {
            e.printStackTrace()
            return null
        }
    }
}